conductr_cli/test/test_conduct_agents.py

```diff
@@ -0,0 +1,172 @@
+                                       timeout=DEFAULT_HTTP_TIMEOUT, headers={'Host': '127.0.0.1'})
+        self.assertEqual(
+            strip_margin("""|ADDRESS  ROLES  OBSERVED BY
+                            |"""),
+            self.output(stdout))
+
+    def test_role_match(self):
+        self.maxDiff = None
+
+        filtered_by_role_web_args = self.default_args.copy()
+        filtered_by_role_web_args.update({'role': 'web'})
+
+        http_method = self.respond_with(text=self.fake_output)
+
+        stdout = MagicMock()
+
+        input_args = MagicMock(**filtered_by_role_web_args)
+        with patch('requests.get', http_method):
+            logging_setup.configure_logging(input_args, stdout)
+            result = conduct_agents.agents(input_args)
+            self.assertTrue(result)
+
+        http_method.assert_called_with(self.default_url, auth=self.conductr_auth, verify=self.server_verification_file,
+                                       timeout=DEFAULT_HTTP_TIMEOUT, headers={'Host': '127.0.0.1'})
+        self.assertEqual(
+            strip_margin("""|ADDRESS                                                                            ROLES     OBSERVED BY
+                            |akka.tcp://conductr-agent@192.168.10.1:2552/user/reaper/cluster-client#-596247188  web,data   1129598726
+                            |akka.tcp://conductr-agent@192.168.10.2:2552/user/reaper/cluster-client#-775189131  web        1129598726
+                            |akka.tcp://conductr-agent@192.168.10.3:2552/user/reaper/cluster-client#1858099110  web,data   1129598726
+                            |"""),
+            self.output(stdout))
+
+    def test_role_match_data(self):
+        self.maxDiff = None
+
+        filtered_by_role_web_args = self.default_args.copy()
+        filtered_by_role_web_args.update({'role': 'data'})
+
+        http_method = self.respond_with(text=self.fake_output)
+
+        stdout = MagicMock()
+
+        input_args = MagicMock(**filtered_by_role_web_args)
+        with patch('requests.get', http_method):
+            logging_setup.configure_logging(input_args, stdout)
+            result = conduct_agents.agents(input_args)
+            self.assertTrue(result)
+
+        http_method.assert_called_with(self.default_url, auth=self.conductr_auth, verify=self.server_verification_file,
+                                       timeout=DEFAULT_HTTP_TIMEOUT, headers={'Host': '127.0.0.1'})
+        self.assertEqual(
+            strip_margin("""|ADDRESS                                                                            ROLES     OBSERVED BY
+                            |akka.tcp://conductr-agent@192.168.10.1:2552/user/reaper/cluster-client#-596247188  web,data   1129598726
+                            |akka.tcp://conductr-agent@192.168.10.3:2552/user/reaper/cluster-client#1858099110  web,data   1129598726
```

Comment:
I think the build failed since `OBSERVED BY` is now an Akka address instead of UID.